packages/harvester/test/MockHarvester.test.ts

```diff
@@ -72,4 +87,62 @@ describe("MockHarvester", () => {
 
     expect(gasUsed).to.be.lte(gasLimit);
   });
+
+  it("should not harvest because of canHarvestVault function", async () => {
+    await harvester.setCanHarvestVault(false);
+    await harvester.setShouldHarvestVault(true);
+    const { upkeepNeeded_ } = await harvester.checkUpkeep([]);
+    await harvester.setCanHarvestVault(true);
+    await harvester.setShouldHarvestVault(true);
+    await harvester.setHarvestVault(true);
```

Comment:
Instead of resetting the state in each test, let's change the `before` block to `beforeEach` and have a new deployment for each case.

Reply:
It's not before each but after. I reset everything to the default values after each test in case something crashes or somebody want to use the deployed contract for something else. This tests are executed in polygon fork that stays operational after the tests, not in the hardhat local chain that is destroyed after the tests.
I will move them in `afterEach` instead. Is it ok?

Reply:
Regardless of whether it's a fork or not, it should deploy a new instance if it's in a `beforeEach` block, afak.